system/Database/BaseBuilder.php

```diff
@@ -2653,6 +2653,105 @@ public function delete($where = '', ?int $limit = null, bool $resetData = true)
         return $this->testMode ? $sql : $this->db->query($sql, $this->binds, false);
     }
 
+    /**
+     * Sets data and calls batchExecute to run queries
+     *
+     * @param array|object|null $set         a dataset or select query
+     * @param array|RawSql|null $constraints
+     *
+     * @return false|int|string[] Number of rows affected or FALSE on failure, SQL array when testMode
+     */
+    public function deleteBatch($set = null, $constraints = null, int $batchSize = 100)
+    {
+        $this->onConstraint($constraints);
+
+        if ($set !== null && $set !== []) {
+            $this->setData($set, true);
+        }
+
+        return $this->batchExecute('_deleteBatch', $batchSize);
+    }
+
+    /**
+     * Generates a platform-specific batch update string from the supplied data
+     */
+    protected function _deleteBatch(string $table, array $keys, array $values): string
+    {
+        $sql = $this->QBOptions['sql'] ?? '';
+
+        // if this is the first iteration of batch then we need to build skeleton sql
+        if ($sql === '') {
+            $constraints = $this->QBOptions['constraints'] ?? [];
+
+            if ($constraints === []) {
+                if ($this->db->DBDebug) {
+                    throw new DatabaseException('You must specify a constraint to match on for batch deletes.'); // @codeCoverageIgnore
+                }
+
+                return ''; // @codeCoverageIgnore
+            }
+
+            $alias = $this->QBOptions['alias'] ?? '_u';
+
+            $sql = 'DELETE ' . $table . ' FROM ' . $table . "\n";
+
+            $sql .= "INNER JOIN (\n{:_table_:}";
+
+            $sql .= ') ' . $alias . "\n";
+
+            $sql .= 'ON ' . implode(
+                ' AND ',
+                array_map(
+                    static fn ($key, $value) => (
+                        $value instanceof RawSql ?
+                    $value :
+                    (
+                        is_string($key) ?
+                    $table . '.' . $key . ' = ' . $alias . '.' . $value :
+                    $table . '.' . $value . ' = ' . $alias . '.' . $value
+                    )
+                    ),
+                    array_keys($constraints),
+                    $constraints
+                )
+            );
```

Comment:
Does this code have the correct indentation? It is terrible to read - or it's just me.
Can you see if anything will scream at you if you try to make it more readable?
I hope it is not forced by CS Fixer...

Reply:
I was able to fix these and CS Fixer didn't revert them.